### PR TITLE
[#4829] improvement(client-python): Sphinx Initialization

### PR DESCRIPTION
**Pull request by noidname01**
### What changes were proposed in this pull request?

* Add Sphinx configuration files and templates for generating documents automatically
* Add new task `doc` for Sphinx documents generation and remove old task for docs
* Generated website files will be in `docs/build/html`, open `docs/build/html/index.html` to see the home page
* Feel free to improve this document configuration (what contents should be included, add project logos in documents, etc.)

#### Current Document Display
image

> NOTES:
> * The current theme is [Furo](https://pradyunsg.me/furo/), feel free to change if other theme is better
> * It still needs a lot of revision, like mentioned in https://github.com/apache/gravitino/issues/4797#issuecomment-2323197095

### Why are the changes needed?

Fix: #4829

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

./gradlew client:client-python:doc

### CODE REVIEW

**Comment by jerryshao** [comment]
@noidname01 can you please fix the conflict?

**Comment by jerryshao** [comment]
Let me take a try on this, to how Sphinx goes compared to pydoc.

File: clients/client-python/docs/source/index.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+-----------
+.. autosummary::
+    :toctree: generated
+    :template: custom-module-template.rst
+    :recursive:
+
+    gravitino
+
+   
+
```
Comment (jerryshao) [inline clients/client-python/docs/source/index.rst:36]: Can we remove the trailing blank lines, are they nesssary?

**Comment by jerryshao** [comment]
Looks like each class will have a Apache License on the doc, maybe sphinx treats the ASF header as a Python doc, is it possible to remove it?

image

**Comment by noidname01** [comment]
@jerryshao Yeah, I plan to do it in next PR originally, as mentioned in #4797, because it will bring a lot of changes.

**Comment by jerryshao** [comment]
Oh, I see. Thanks for the explanation.

File: clients/client-python/docs/source/index.rst (hunk 1 above)
Comment (jerryshao) [reply clients/client-python/docs/source/index.rst:36]: @noidname01 can you please check this comment?

Comment (noidname01) [reply clients/client-python/docs/source/index.rst]: Fixed!

**Review by jerryshao — APPROVED** [review]

**Pull request merged**